compile openssl failed: undefined reference to 'timespecsub'

Since https://github.com/libressl-portable/openbsd/commit/f4017f42493f468fc323a37093edefb9a0f24020
```
apps_posix.o:apps_posix.c:function app_timer_real: error: undefined reference to 'timespecsub'
```
build Linux 4.14.0-1-amd64 with gcc 7.2.1

--- a/include/compat/time.h
+++ b/include/compat/time.h
@@ -21,3 +21,15 @@ struct tm *__gmtime_r(const time_t * t, struct tm * tm);
 #ifndef HAVE_TIMEGM
 time_t timegm(struct tm *tm);
 #endif
+
+#ifndef timespecsub
+#define timespecsub(tsp, usp, vsp)                                      \
+        do {                                                            \
+                (vsp)->tv_sec = (tsp)->tv_sec - (usp)->tv_sec;          \
+                (vsp)->tv_nsec = (tsp)->tv_nsec - (usp)->tv_nsec;       \
+                if ((vsp)->tv_nsec < 0) {                               \
+                        (vsp)->tv_sec--;                                \
+                        (vsp)->tv_nsec += 1000000000L;                  \
+                }                                                       \
+        } while (0)
+#endif

